### PR TITLE
V3: Improving styling to match Home Assistant

### PR DESCRIPTION
**Pull request by kevireilly**
With the addition of packaged workspace building of multiple versions (#53) as a prerequisite for revising the ESPHome device pages to look more like Home Assistant (#49), this pull request is rebase of that work to a new v3 version.
- The HA styling revision that was previously a modification of v2 has been forked over to a new v3 package
- The original v2 has been restored to its former state that is out in the world currently
- Removed the `vfallback` work for now and did some small clean up and verification/adjustment for deployment

Take v3 out for a spin:
```yaml
web_server:
  port: 80
  js_url: https://deploy-preview-54--esphome-webserver.netlify.app/v3/www.js
```

This pull request supersedes the original pull request #49:
> This PR is a styling upgrade to match the styling of HomeAssistant entities card.
> 
> * icons using the standard HA iconset via on demand and thus with a graceful degradation (no icon shown if WAN is down)
> * group by entity_categories
> * instead of table use div and grid/flex layouts
> * added uptime based on ping value
> * color picker for rgb lights
> * Tag column hidden on web view
> * update icon to use the
> 
> Icons and entity_categories functionally needs matching PR in esphome/web_server component - though the two still work in the previous way without the other.
> 
> Not sure about coding standards, or formatting in this repo. Please comment and help me with a pointer if styling changes are needed.
> 
> Related discord discussion - https://discord.com/channels/429907082951524364/1177062706340171807

Closes #49

### CODE REVIEW

File: packages/v2/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "xbuild": "vite build --emptyOutDir",
     "build": "vite build --emptyOutDir && npm run deploy",
     "serve": "vite preview",
-    "deploy": "bash -c '../../scripts/make_header.sh ../../_static/v2 server_index.h web_server'"
+    "deploy": "bash -c '../../scripts/make_header.sh ../../_static/v2 server_index.h web_server INDEX_GZ'"
```
Comment (kevireilly) [inline packages/v2/package.json:10]: I adapted this change from the original pull request, but I am not familiar with what it is doing or why it is required.

File: packages/v3/src/main.css [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+  line-height: 1.5;
+  word-wrap: break-word;
+}
+
+button {
+  cursor: pointer;
+}
+*/
+
+
```
Comment (kevireilly) [inline packages/v3/src/main.css:284]: Thinking all of the commented code above can probably be removed at this point?

File: packages/v3/vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        fileName: ".gz",
+      }),
+      enforce: "post",
+      apply: "build",
+    },
+    // {
+    //   ...gzipPlugin({ filter: /\.(js|css|html|svg)$/ }),
+    //   enforce: "post",
+    //   apply: "build",
+    // }
```
Comment (kevireilly) [inline packages/v3/vite.config.ts:56]: @afarago Why was this removed? Seems like it is resulting in the following files not being created:
- www.js.br
- index.html.br

File: scripts/make_header.sh [hunk 4]
```diff
@@ -8,7 +8,7 @@ namespace esphome {
 namespace $3 {
 
 EOT
-echo "const uint8_t INDEX_GZ[] PROGMEM = {" >> ./$1/$2
+echo "const uint8_t $4[] PROGMEM = {" >> ./$1/$2
```
Comment (kevireilly) [inline scripts/make_header.sh:11]: @afarago Was this needed for the vfallback work and can this potentially be removed?

**Review by kevireilly — COMMENTED** [review]
I'm not particularly familiar with most of the changes being proposed, but I have ensured that they are exactly like they were as sourced from the original pull request.

File: scripts/make_header.sh (hunk 4 above)
Comment (jesserockz) [reply scripts/make_header.sh:11]: I would think yes, lets revert this change for this PR

File: packages/v3/public/home.svg
Comment (afarago) [inline packages/v3/public/home.svg:1]: `<path>`
should have a self-closing tag ending
`<path .... />

Comment (afarago) [reply packages/v3/public/home.svg:1]: btw: do we even need this file at all? Does not seem to be used anywhere...

**Review by afarago — APPROVED** [review]
Thank you for your work to merge this change in!
Code seems to be OK, build seems to be working for all my devices nicely.

File: packages/v3/public/home.svg
Comment (kevireilly) [reply packages/v3/public/home.svg:1]: Good call and as it does not appear to be in use, I've removed it. I suppose there is a possibility that despite it not being in use directly from this project, that there may be external use cases that expect this hosted file to be here. Since v3 is new and unused in the wild, I'll leave the v2 version intact for the potential scenario of an external use case and we'll see if an external v3 problem pops up.

**Review by jesserockz — APPROVED** [review]

**Comment by DocBrown101** [comment]
From which version will it be available as v3 in an esphome release?
